### PR TITLE
Review and correct various styleguide sections

### PR DESCRIPTION
**Pull request by dispera**
Scope
=====

Continue to review and provide feedback for the styleguide as part of my onboarding process: chevah/onboarding#23,

This review specifically covers Bash/CSS/Dependencies/HTML/JavaScript/References/Testing sections.

Changes
=======

Added improved/correct Markdown syntax highlighting format to CSS, HTML, Javascript and BASH pages.

Replaced:
.. sourcecode:: javascript
by:
\```javascript

In many cases, the highlighting was missing, but in cases like BASH page, the 'sourcecode::'' version was used,
and on my tests, the version I used shows better highlighting (eg. uses colors instead of just making if/elif/else bold in BASH).
In some cases, the 'sourcecode::'' version which was in use, did practically nothing to the code.

I tested using [Dillinger](http://dillinger.io) online Markdown page.

Also reviewed and corrected gramatical errors on the following pages, and tried to improve some bits here and there:
('testing' section is in progress, reviewed up to 'Unit testing')
	modified:   content/bash.rst
	modified:   content/css.rst
	modified:   content/dependencies.rst
	modified:   content/html.rst
	modified:   content/javascript.rst
	modified:   content/references.rst
	modified:   content/testing.rst

Reviewers
===============================

reviewers: @adiroiban

### CODE REVIEW

**Comment by adiroiban** [comment]
> Also reviewed and corrected gramatical errors on the following pages, and tried to improve some bits here and there:

No need to mention each page... the GitHub PR diff already provide that information

The PR description should be about those things which can't be found in any other place.

A PR with a short description is a well written PR ... you don't need to have a huge description for a PR

File: content/bash.rst [hunk 1]
```diff
@@ -10,14 +10,14 @@ General
...
+  For every other script we should use Python.
 
-* If any, public scripts should use ``/bin/sh``. Bash is not available on all
+* Public shell scripts should use ``/bin/sh``. Bash is not available on all
   Unix systems and ksh or other shells are not compatible with Bash syntax.
 
 * Use ``printf`` instead of ``echo`` when using  ``/bin/sh`` as a shell,
   because ``echo`` behaviour is incompatible in Bash and Korn shells.
-  Eg. use ``printf`` instead of ``echo -n`` and ``printf "blabla\n"`` instead
+  Eg. use ``printf "\n"`` instead of ``echo -n`` and ``printf "blabla\n"`` instead
```
Comment (adiroiban) [inline content/bash.rst]: this this ok? does `printf "\n"` has the same effect as `echo -n` ?!

File: content/bash.rst [hunk 2]
```diff
@@ -52,10 +52,9 @@ General
 
 * Always guard your Bash scripts from unexpected errors by using
 
-.. sourcecode:: bash
-
+```bash
```
Comment (adiroiban) [inline content/bash.rst]: is this valid RST formatting?

**Comment by adiroiban** [comment]
I left a few comments, but I am waiting for the PR to be ready... don't forget to set a easy to read title.

Thanks!

File: content/bash.rst (hunk 2 above)
Comment (dispera) [inline content/bash.rst]: No, I mistakenly assumed '.rst' files were as the '.md' files but for git websites, as ReStructured Text shares similarities with Markdown. I will quickly search and correct all the occurrences with the original .. sourcecode::.
Learnt a new language I guess :P

I found an online rst tester at [rst.ninjs.org](http://rst.ninjs.org) so I will compare with that, my comparison with Markdown was wrong.

File: content/bash.rst (hunk 1 above)
Comment (dispera) [inline content/bash.rst]: Not at all, quite the opposite as 'echo -n' removes the trailing newline. 
My mistake - I tested what I wanted to rewrite then, but I must have modified it in the end without a last test. Will be more careful, sorry!!

**Comment by dispera** [comment]
Thanks Adi, I will review tomorrow and correct the PR.

**Comment by dispera** [comment]
I performed the corrections from yesterday's observations, @adiroiban please review when you have time :)

**Comment by adiroiban** [comment]
changes look good. thanks.

@hcs0  Can you please take a look at the changes? Thanks!

**Comment by hcs0** [comment]
@adiroiban @dispera Read through it - looks good also. Thanks

**Comment by adiroiban** [comment]
ok, changes-approved.

feel free to merge

thanks!

**Pull request merged**